pkg/pillar/scripts/device-steps.sh

```diff
@@ -164,6 +167,25 @@ if [ ! -d $PERSISTDIR/log ]; then
     mkdir $PERSISTDIR/log
 fi
 
+
+LAST_EVE_VERSION=""
+if [ -f $LAST_IMAGE_VERSION_FILE ];
```

Comment:
Given that the state transition diagram in these types of converters tends to get really messy, I would suggest we don't have ANY logic in device-steps at all.

In fact, I would suggest we simply *always* invoke converter on boot and let it figure out what to do next.

I also suggest that we allow for it to retur error and thus kill this EVE's boot sequence.

Should we discuss this on the call @kalyan-nidumolu ?

Reply:
That was my original Idea / Change - Always invoke upgradeconverter and let it figure out if any processing is needed. You wanted it to be invoked conditionally - only if needed :-)

I am fine with both ways.. Now that I have already made the changes, running upgradeconverter only on upgrades makes sense too :-)